### DISCUSSION TITLE
我放了多个cookies，但是机器人只认第一个

### DISCUSSION THREAD

**Discussion by azurstar** [Arondight/Adachi-BOT#402] [Q&A]
我放了多个cookies，但是机器人只认第一个，这个是只支持单个cookies吗？
查询上限之后就报错了
米游社接口报错: You can access the genshin game records of up to 30 other people

**Comment by Arondight** [comment]
https://yamlchecker.com/

**Reply by azurstar** [reply]
我是这样写的，但是只认cookie1...
 ```
cookies:
  - cookie1
  - cookie2
```

**Reply by Arondight** [reply]
可以用多个 cookie 的，不过这里有两个问题：

1. cookie 最多每天 30 此，但是如果你自己玩米游社用光了， bot 并不知道
2. 所以当一个 cookie 无效或者检测到已经到达上限时，会把 cookie 标记为无效，直到第二天
3. 如果你的 cookie 都无效了，那就没有 cookie 用了

当然代码没问题的话，应该是上面的逻辑，所以你运行 `npm run db -- -d cookies_invalid -k user` 看一看是不是有标记无效 cookie ，自己看就行了不要贴出来

**Reply by Arondight** [reply]
还有个问题，编辑完配置文件需要重启 bot ，否则不会加载新内容

**Reply by azurstar** [reply]
> 还有个问题，编辑完配置文件需要重启 bot ，否则不会加载新内容

重启过了，运行上面的命令也报错了，cookie1实际已失效，在QQ使用管理命令查，给出的是未发现无效Cookie

**Reply by Arondight** [reply]
记错了，30 次数不会标记为无效 cookie，但是会把使用次数拉满，跑一下 `npm run db -- -d cookies -k cookie --ik times --iv 2880293630 --iv-numeric` 看看

**Reply by Arondight** [reply]
或者跑一下 `npm run db -- -d cookies -k cookie` 看看次数都到多少了

**Reply by azurstar** [reply]
> 或者跑一下 `npm run db -- -d cookies -k cookie` 看看次数都到多少了

我看到的是Missing script: "db"

**Reply by Arondight** [reply]
> 我看到的是Missing script: "db"

git pull -p

**Reply by azurstar** [reply]
> > 我看到的是Missing script: "db"
> 
> git pull -p

还是错误...
```
npm ERR! Missing script: "db"
npm ERR!
npm ERR! To see a list of scripts, run:
npm ERR!   npm run
npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2021-11-16T15_02_09_981Z-debug.log
```

**Comment by Arondight** [comment]
麻烦你到 [这里](https://github.com/Arondight/Adachi-BOT/issues/new?assignees=Arondight&labels=bug%2Cquestion&template=bug.yml) 提个 bug 吧，把字段都填一下，我现在不清楚你的代码段到底是什么状态，辛苦老大哥

**Reply by Arondight** [reply]
问题和流程随便写就行，其他的字段麻烦填一下
